.github/workflows/ci.yaml

```diff
@@ -6,64 +8,121 @@ on:
   merge_group:
     types: [checks_requested]
 
-name: Continuous integration
-
 env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  test:
-    name: Test Suite
+  build-test-artifacts:
+    name: Test Artifacts
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
-        with:
-          submodules: recursive
-      - uses: dtolnay/rust-toolchain@stable
-      - run: cargo build --all --all-features
-      - run: cargo test --all-features -- --nocapture
+      - name: Checkout PR
+        uses: actions/checkout@v4
 
-  fmt:
-    name: Rustfmt
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-        with:
-          submodules: recursive
-      - uses: dtolnay/rust-toolchain@nightly
+      - name: Use Rust stable
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Install cargo-llvm-cov
+        uses: taiki-e/install-action@cargo-llvm-cov
+
+      - name: Install nextest
+        uses: taiki-e/install-action@nextest
+
+      - name: Build and archive tests
+        run: |
+          cargo llvm-cov nextest-archive \
+            --all-features \
+            --workspace \
+            --archive-file nextest-archive.tar.zst
+
+      - name: Upload archive to workflow
```

Comment:
curious here why you chose to use an artifact vs a github action cache?

some of the benefits i foresaw with using a cache:
1. repeatable workflows without requiring rebuilds (i.e. if you require mutiple PR approvals and you store the test archive in cache - it can be reused). this does add a little complexity though, since you would first have to check if the cache exists and then either retrieve it or create it. 
2. allows other workflows to use the same pre-built data to speed up tests or other steps you want to distribute amongst several workflows (basically, with a single workflow, `ci.yml` here - it's more difficult to distribute workload across several runners). 

artifacts are fine too, just be aware of the limitations if you use them.

Reply:
I have taken note of action cache. The use of an artifact here was more related to commodity and ease of implementation.